tests/unit/btree_stress_test.c

```diff
@@ -266,25 +277,138 @@ CTEST2(btree_stress, test_random_inserts_concurrent)
       (cache *)&data->cc, &data->dbtree_cfg, packed_root_addr, nkvs, data->hid);
    ASSERT_NOT_EQUAL(0, rc, "Invalid ranges in packed tree\n");
 
+   // Release memory allocated in this test case
+   for (uint64 i = 0; i < nthreads; i++) {
+      platform_free(data->hid, params[i].scratch);
+   }
+   platform_free(hid, params);
+}
+
+/*
+ * -------------------------------------------------------------------------
+ * Test case to exercise random inserts of large volumes of data, and then
+ * invoke some BTree-print methods, to verify that they are basically working.
+ * The initial work to setup the BTree and load some data is shared between
+ * this and the test_random_inserts_concurrent() sub-case.
+ * -------------------------------------------------------------------------
+ */
+CTEST2(btree_stress, test_btree_print_diags)
+{
+   int nkvs     = 1000000;
+   int nthreads = 8;
+
+   mini_allocator mini;
+
+   uint64 root_addr = btree_create(
+      (cache *)&data->cc, &data->dbtree_cfg, &mini, PAGE_TYPE_MEMTABLE);
+
+   platform_heap_id      hid    = platform_get_heap_id();
+   insert_thread_params *params = TYPED_ARRAY_ZALLOC(hid, params, nthreads);
+
+   load_thread_params(params,
+                      nthreads,
+                      data->hid,
+                      (cache *)&data->cc,
+                      &data->dbtree_cfg,
+                      &mini,
+                      root_addr,
+                      nkvs);
+
+
+   platform_status ret;
+   ret = do_n_thread_creates(
+      "insert thread", params, nthreads, data->ts, data->hid, insert_thread);
+   ASSERT_TRUE(SUCCESS(ret));
+
+   for (uint64 thread_no = 0; thread_no < nthreads; thread_no++) {
+      platform_thread_join(params[thread_no].thread);
+   }
+
+   uint64 packed_root_addr = pack_tests(
+      (cache *)&data->cc, &data->dbtree_cfg, data->hid, root_addr, nkvs);
+   if (0 < nkvs && !packed_root_addr) {
+      ASSERT_TRUE(FALSE, "Pack failed.\n");
+   }
+
    // Exercise print method to verify that it basically continues to work.
+   CTEST_LOG_INFO("\n**** btree_print_tree() on BTree root=%lu****\n",
+                  packed_root_addr);
    btree_print_tree(Platform_default_log_handle,
                     (cache *)&data->cc,
                     &data->dbtree_cfg,
                     packed_root_addr);
 
+   uint64 meta_page_addr =
+      btree_root_to_meta_addr(&data->dbtree_cfg, packed_root_addr, 0);
+   CTEST_LOG_INFO("\n**** mini_keyed_print() BTree root=%lu"
+                  ", meta page addr=%lu ****\n",
+                  packed_root_addr,
+                  meta_page_addr);
+
+   // Exercise print method of mini-allocator's keyed meta-page
+   mini_keyed_print(
```

Comment:
This is a new invocation of this print method to print BTree metadata page contents.

This invocation seems to work right, too, and prints what seem like reasonable content:

```
**** mini_keyed_print() BTree root=1326710784, meta page addr=1326714880 ****
---------------------------------------------------------------------
| Mini Keyed Allocator -- meta_head:   1326714880                   |
---------------------------------------------------------------------
| meta addr:   1326714880 (2)                                       |
---------------------------------------------------------------------
| idx | batch |    extent_addr |          start_key |  rc |
---------------------------------------------------------------------
|   0 |     0 |     1326841856 | 0x00000d093a9a1700 |   2 |
|   1 |     0 |     1327890432 | 0x000742d1b6583900 |   2 |
|   2 |     1 |     1326972928 | 0x00000d093a9a1700 |   2 |
|   3 |     0 |     1328021504 | 0x00137c3e4b314000 |   2 |
|   4 |     0 |     1328283648 | 0x001df84b01da2a00 |   2 |
|   5 |     0 |     1328414720 | 0x0026020907020200 |   2 |
|   6 |     0 |     1328545792 | 0x0031fa2efde54b00 |   2 |
|   7 |     0 |     1328676864 | 0x0039713e18992a00 |   2 |
|   8 |     0 |     1328807936 | 0x0044cc7bbd592300 |   2 |
|   9 |     0 |     1328939008 | 0x005006e951322a00 |   2 |
[...]
```

and so on.